### PR TITLE
Add the ability to initialise a compute project from a specific branch

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
As requested by @katie-martin-fastly, this adds the ability to clone from a specific template repository branch when initialising a new Compute@Edge project via `compute init`. We then use this logic to ensure that the default template is pointing to a named branch. This ensures we maintain compatibility with the `fastly` guest crate API.

### Note
- I'm keeping the `--branch` flag hidden for the time being as we don't want to add confuse to the normal path of supplying templates repo urls. 
- I also cleaned up the `compute init` tests and fixed a rendering issue with author, but are small enough changes that don't warrant their own PR.

### CODE REVIEW

File: pkg/compute/init.go [hunk 1]
```diff
@@ -68,6 +71,7 @@ func NewInitCommand(parent common.Registerer, globals *config.Data) *InitCommand
 	c.CmdClause.Flag("description", "Description of the package").Short('d').StringVar(&c.description)
 	c.CmdClause.Flag("author", "Author of the package").Short('a').StringVar(&c.author)
 	c.CmdClause.Flag("from", "Git repository containing package template").Short('f').StringVar(&c.from)
+	c.CmdClause.Flag("from-branch", "Git branch name to clone from package template repository").Hidden().StringVar(&c.fromBranch)
```
Comment (peterbourgon) [inline pkg/compute/init.go:74]: This feels like it overloads `--from`. What about `--branch` or `--remote-branch`?

Comment (phamann) [reply pkg/compute/init.go]: Resolved in https://github.com/fastly/cli/pull/14/commits/541d1419a04dcd60e00d52a7294036342380db06

File: pkg/compute/init.go [hunk 2]
```diff
@@ -259,10 +263,19 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
 	}
 	defer os.RemoveAll(tempdir)
 
+	var ref plumbing.ReferenceName
+	if c.from == defaultTemplate {
+		ref = plumbing.NewBranchReferenceName(defaultTemplateBranch)
+	}
+	if c.branch != "" {
+		ref = plumbing.NewBranchReferenceName(c.branch)
+	}
```
Comment (peterbourgon) [inline pkg/compute/init.go:272]: Does `ref` need a non-blank default?

Comment (phamann) [reply pkg/compute/init.go:272]: I tested this and read a bit of the go-git source, without you get the HEAD reference which was the same behaviour before this change.
